### PR TITLE
line chart: make threejs coordinator match DOM size

### PR DESCRIPTION
**Pull request by stephanwlee**
Previously, we had a camera coordinate system where we had a fixed camera
view point. This was great because irrespective of the actual DOM size,
we could keep the objects position static. However, this approach has a
major problem. Now that we want to render shapes like triangle and
circle, geometry that is defined in the camera coordinate system has to
account for mismatch in camera coordinate aspect ratio vs. DOM camera
coordinate aspect ratio.

We want circle that is not an oval in browser pixel location. To do
that, when creating a geometry in the camera coordinates, we need to
account for discrepancy in aspect ratio, which generally is harder than
just matching the camera view point to that of the DOM size. As such,
this change removes the camera coordinate system separate from DOM but
supports the y-axis flipping which is the only remaining difference in
the coordinate systems we support.

### CODE REVIEW

File: tensorboard/webapp/widgets/line_chart_v2/lib/coordinator.ts [hunk 1]
```diff
@@ -102,7 +121,9 @@ export class Coordinator {
       ),
       this.yScale.forward(
         domain.y,
-        [rect.y + rect.height, rect.y],
+        this.getYAxisDirection() === 1
+          ? [rect.y, rect.y + +rect.height]
```
Comment (psybuzz) [inline tensorboard/webapp/widgets/line_chart_v2/lib/coordinator.ts:125]: Is the 2nd '+' needed?  `rect.height` already seems to be a number

File: tensorboard/webapp/widgets/line_chart_v2/lib/coordinator_test.ts [hunk 2]
```diff
@@ -159,7 +159,7 @@ describe('line_chart_v2/lib/coordinator test', () => {
     });
 
     describe('#transformDataToUiCoord', () => {
-      beforeEach(() => {
+      it('converts with y-axis pointing up', () => {
```
Comment (psybuzz) [inline tensorboard/webapp/widgets/line_chart_v2/lib/coordinator_test.ts:162]: Maybe I'm confusing myself, but this test uses the default coordinator with getYAxisDirection -1.  Should the text say "converts with y-axis pointing down"?

File: tensorboard/webapp/widgets/line_chart_v2/lib/coordinator.ts [hunk 3]
```diff
@@ -60,6 +61,24 @@ export class Coordinator {
...
+   * |
+   * | viewBoxYDirection = -1
+   * ↓
+   *
+   * +1: Suitable in 3-D aware context where you have full cartesian coordinates.
+   * -1: Suitable for DOM where <0, 0> is top-left corner of a DOM.
+   */
+  getYAxisDirection(): -1 | 1 {
+    return -1;
+  }
```
Comment (psybuzz) [inline tensorboard/webapp/widgets/line_chart_v2/lib/coordinator.ts:80]: I really do like the helpful ASCII comment, it's really clear!  An optional alternative is to name this `isYAxisPointedDown(): boolean`, which may help when reading callsites without having to go back to the ascii for reference.

**Review by psybuzz — APPROVED** [review]
lgtm, the description makes a lot of sense.  Also I didn't realize TS allowed return types `-1 | 1`!

File: tensorboard/webapp/widgets/line_chart_v2/lib/coordinator_test.ts (hunk 2 above)
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/coordinator_test.ts:162]: This is for the ThreeCoordinator where y-axis is pointing up (L142 reads: `describe('ThreeCoordinator', () => {`)

Comment (psybuzz) [reply tensorboard/webapp/widgets/line_chart_v2/lib/coordinator_test.ts:162]: Ah, gotcha.  Silly me for not scrolling up to read around this line!

**Pull request merged**